components/script/dom/serviceworkerregistration.rs

```diff
@@ -147,4 +154,9 @@ impl ServiceWorkerRegistrationMethods for ServiceWorkerRegistration {
     fn UpdateViaCache(&self) -> ServiceWorkerUpdateViaCache {
         self.update_via_cache
     }
+
+    // https://w3c.github.io/ServiceWorker/#service-worker-registration-navigationpreload
+    fn NavigationPreload(&self) -> DomRoot<NavigationPreloadManager> {
+        self.navigation_preload.get().unwrap()
```

Comment:
This can use `or_init` to lazily create the value instead of doing it in the constructor. This can save a little bit of memory when the NavigationPreloadManager is never accessed.